internal/statesync/reactor.go

```diff
@@ -247,6 +247,10 @@ func (r *Reactor) OnStop() {
 	// wait for any remaining requests to complete
 	<-r.dispatcher.Done()
 
+	r.syncer.Close()
+
+	<-r.dispatcher.Done()
```

Comment:
Did you mean to remove the same at 248? I'm not sure what this is trying to do: Once call to `r.dispatcher.Close()` closes its channel (https://github.com/tendermint/tendermint/blob/c9d92f5f19f75500c6a4b73b8ec04fe9dd8abb0d/internal/statesync/dispatcher.go#L144), which is an unbuffered channel that is only closed, this will return immediately always.  Am I missing something else?

Reply:
Nope, you are missing nothing! I was basically duplicating the pattern from dispatcher and made a copy+paste error. Fixed here, sorry for the mistake. There's really no need for the done here, I can remove.